Target.cc

```diff
@@ -429,8 +431,8 @@ const char *Target::NameIP(char *buf, size_t buflen) const {
 
 /* This next version returns a static buffer -- so no concurrency */
 const char *Target::NameIP() const {
-  if (!nameIPBuf) nameIPBuf = (char *) safe_malloc(MAXHOSTNAMELEN + INET6_ADDRSTRLEN);
-  return NameIP(nameIPBuf, MAXHOSTNAMELEN + INET6_ADDRSTRLEN);
+  if (!nameIPBuf) nameIPBuf = (char *) safe_malloc(FQDN_LEN + INET6_ADDRSTRLEN);
+  return NameIP(nameIPBuf, FQDN_LEN + INET6_ADDRSTRLEN);
```

Comment:
NameIP adds 3 characters by putting the hostname and IP string into the format string `"%s (%s)"`. So this should be `FQDN_LEN + INET6_ADDRSTRLEN + 3`, probably with a comment stating why.